[vcpkg] Add `vcpkg_install_copyright()` portfile function (#25239)
microsoft/vcpkg: docs/maintainers/vcpkg_install_copyright.md

> # vcpkg_install_copyright

Reviewer: Please take a look at https://github.com/microsoft/vcpkg/pull/25255/files for how helper docs should be formatted moving forward

Author: I think docs have correct formatting now. I will push my changes once CI is done.

Reviewer 2: I think you should format the doc now.

Author: Did I miss anything? I already formatted the file.

The latest version of this document lives in the [vcpkg repo](https://github.com/Microsoft/vcpkg/blob/master/docs/maintainers/vcpkg_install_copyright.md).

Merges multiple copyright files into a single file and install it.

> ## Usage Example:

Reviewer: ```suggestion
## Usage
```

```cmake

> vcpkg_install_copyright(FILE_LIST "${SOURCE_PATH}/LICENSE/license.md" "${SOURCE_PATH}/LICENSE/license_gpl.md"...)

Reviewer: The examples should be moved to the end in an `Examples` section. In this first `Usage` section, the exact and complete set of options should be listed. Something like:
```cmake
vcpkg_install_copyright(FILE_LIST <file>...)
```

```

You can also collect the required files using a `GLOB` pattern:

```cmake
file(GLOB LICENSE_FILES "${SOURCE_PATH}/LICENSES/*")
vcpkg_install_copyright(FILE_LIST ${LICENSE_FILES})
```

## Parameters:

### FILE_LIST

> Specifies a list of license files with absolute paths. You must provide at least 2 files.

Reviewer: ```suggestion
Specifies a list of license files with absolute paths. At least one file must be provided.
```

I think the code should change to allow a single file. In the case of a single file, it should be equivalent to:
```cmake
configure_file("${FILE}" "${CURRENT_PACKAGES_DIR}/share/${PORT}/copyright" COPYONLY)
```

Author: The problem is that we would have to say "every port must be changed to use `vcpkg_install_copyright` instead". I don't know if this was a good idea.

Reviewer 3: > I think the new name `vcpkg_concat_copyright` makes clear that this function can't be used for a single file because concatenating a single file doesn't make any sense.

Well, even for a single file, it would allow to abstract away the target file name and path, and allow for comments.

Author: > Well, even for a single file, it would allow to abstract away the target file name and path, and allow for comments.

We have to keep in mind what the idea behind this function was: concatenating multiple license files. 

You suggest that we use this to in general install copyright file(s). This is a different objective and we have to agree on what we actually want:

1. Replace the current way of installing a copyright file
2. Concatenate multiple copyright files and add comments
    Maybe there should be a separate function for adding comments to a single copyright file.
    We might also want to have a function to extract copyright information out of other files, e.g. `readme`.
3. Just concatenate multiple copyright files and add comments at the top

If you want to install just a single license file, please use

```cmake
file(INSTALL "${SOURCE_PATH}/LICENSE" DESTINATION "${CURRENT_PACKAGES_DIR}/share/${PORT}" RENAME copyright)
```

## Notes

> This function concatinates the contents of multiple copyright files to a single file.

Reviewer: ```suggestion
This function concatenates the contents of multiple copyright files to a single file.
```
